share/cmake/modules/Findyaml-cpp.cmake

```diff
@@ -118,51 +124,35 @@ if(NOT OCIO_INSTALL_EXT_PACKAGES STREQUAL ALL)
         if(PC_yaml-cpp_FOUND)
             set(yaml-cpp_VERSION "${PC_yaml-cpp_VERSION}")
         endif()
+
+        # Create target
+        add_library(yaml-cpp::yaml-cpp UNKNOWN IMPORTED GLOBAL)
+        set_target_properties(yaml-cpp::yaml-cpp PROPERTIES
+            IMPORTED_LOCATION ${yaml-cpp_LIBRARY}
+            INTERFACE_INCLUDE_DIRECTORIES ${yaml-cpp_INCLUDE_DIR}
+        )
+        # Required because Installyaml-cpp.cmake creates `yaml-cpp::yaml-cpp`
+        # as an alias, and aliases get resolved in exported targets.
+        add_library(yaml-cpp ALIAS yaml-cpp::yaml-cpp)
```

Comment:
Not sure I understand exactly the issue, would it be possible to fix Installyaml-cpp instead?

Reply:
Installyaml-cpp uses `ExternalProject_Add`, so you have to deal with whatever your get from that project. The most sensible alternative to the workaround here is to upgrade the version that gets installed to yaml-cpp 0.8.

Reply:
When Installyaml-cpp is called, the Findyaml-cpp is not called after, unless I'm missing something? Is this about the static library build? Looking at the Installyaml-cpp it also always will create a yaml-cpp::yaml-cpp not matter the library version requested isn't it?

Reply:
I apologies about the last part of the answer, I was looking at my own local changes, I'd suggest to update the Installyaml-cpp script to create a yaml-cpp::yaml-cpp target for simplicity?

Reply:
You weren't wrong, there is an alias target in `Installyaml-cpp.cmake`. But there is one case where that doesn't help: The generated `OpenColorIOTargets.cmake` will nonetheless link the static library to `yaml-cpp` instead of `yaml-cpp::yaml-cpp`.

This was actually caught by "Test CMake Consumer with static OCIO" CI task.

Reply:
Ok thanks for the details, I'm fine using this if needed then. I tested on my [fork](https://github.com/remia/OpenColorIO/pull/51) with similar changes but didn't get the error on the CMake Static Consumer, any idea why?

It seems like the yaml-cpp upstream have an ongoing [issue](https://github.com/jbeder/yaml-cpp/issues/1241) that might be related to this situation. Looking at the code, it looks like yaml0-cpp is still the target defined upstream and yaml-cpp::yaml-cpp is just an alias even in 0.8.0. Looking at homebrew 0.8.0 version, I have `set(YAML_CPP_LIBRARIES "yaml-cpp")`,

Reply:
> I tested on my https://github.com/remia/OpenColorIO/pull/51 with similar changes but didn't get the error on the CMake Static Consumer, any idea why?

Sorry, no. I was able to replicate that problem by following the steps taken in the CI job.

The upstream issue was actually fixed in https://github.com/jbeder/yaml-cpp/pull/1212.